content/articles/dealing-with-python's-missing-values-a-complete-guide/index.md

```diff
@@ -0,0 +1,105 @@
+
+Using linear regression as an example, an error can be made. As soon as we know what went wrong with the model, we have to figure out how to deal with the lack of data until we get a clearer grasp of what went wrong in the first place. If the reaction is positive, we'll proceed. If it is negative, we'll stop.
+```Python
+dataset["AgeNumber"][:10]
+```
+### Removing the rows/columns that are not in use
+The next simplest strategy is to exclude observations with missing data. In the end, you can miss out on important pieces of information. Python's pandas module has a dropna() method for getting rid of empty columns. When dealing with machine learning issues, rely on your own experience or that of a domain expert rather than attempting to fill in every blank in every column.
+
+There are both advantages and disadvantages to removing the rows/columns; for example, the model becomes more robust after deleting missing data. If you have a large amount of missing data, your model will be less accurate.
+```Python
+dataset.dropna(inplace=True)
+print(dataset.isnull().sum())
+```
+### Imputation based on the mean
+After calculating the non-missing values in a column, each missing value can be restored individually and independently. This tool is severely limited if you want to use it with anything other than numbers. If you only have a small amount of data, it's a quick and straightforward way to analyze it The lack of feature correlations is one of the flaws, but there are others as well. One column at a time can be used with this technique, and that's it. The model is likely to employ a skewed mean value to make up for the loss of the outlier treatment.
+
+Imputation based on the mean has a number of drawbacks, including the fact that technique only works with numerical datasets and fails when independent variables are correlated.
+```Python
+dataset["AgeNumber"] = dataset["AgeNumber"].replace(num.NaN, dataset["AgeNumber"].mean())
+print(dataset["AgeNumber"][:10])
+```
+### Using the Median to Compute
+The median values of the population can be used to deal with outliers in the prior procedure. When sorting, a column's center value is updated rather than an outlier. No correlation between the independent variables was found, and it only works with numerical datasets.
+```Python
+dataset["AgeNumber"] = dataset["AgeNumber"].replace(num.NaN, dataset["AgeNumber"].median())
+print(dataset["AgeNumber"][:10])
+```
+### Imputation  based on the most common values
+It can be applied to categorical variables with a restricted number of values. You can use the drop-down menu to select the most common value. Education level is a good example of an ordinal categorical attribute that falls into this category: Data bias can occur since feature relationships are not considered when utilizing this procedure. If the category values are not evenly distributed among the categories, the likelihood of biasing the data increases.

```

Comment:
`You can use the drop-down menu to select the most common value.`

Drop down menu? Where did we get this from?